gdx/src/com/badlogic/gdx/graphics/g2d/ExpandableSprite.java

```diff
@@ -0,0 +1,804 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.graphics.g2d;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.VertexAttribute;
+import com.badlogic.gdx.graphics.VertexAttributes.Usage;
+import com.badlogic.gdx.graphics.glutils.ShaderProgram;
+import com.badlogic.gdx.math.MathUtils;
+import com.badlogic.gdx.math.Rectangle;
+import com.badlogic.gdx.utils.IntMap;
+import com.badlogic.gdx.utils.NumberUtils;
+
+/** Behaves similarly to {@linkplain Sprite}, but it can be instantiated with extra vertex attributes beyond the standard ones
+ * (position, color, and texture coordinates). The sprite can be drawn with a SpriteBatch that was instantiated with the same
+ * array of VertexAttributes that were defined in the input {@linkplain Template}. See
+ * {@link SpriteBatch#SpriteBatch(int, ShaderProgram, VertexAttribute...)}.
+ * <p>
+ * Attempting to draw the Sprite with a SpriteBatch whose extra attributes don't match will cause unexpected rendering and
+ * possible Exceptions. */
+public class ExpandableSprite extends TextureRegion {
```

Comment:
Agreed. I'm trying out a simpler version that extends Sprite and has no impact on the current Sprite code. It keeps the extra attribute data in a separate array from the base vertices and (unfortunately) collates the two arrays when drawing. The code will be far simpler--not only does it eliminate all the copy-paste, but also the AttributesMapping inner class.